### PR TITLE
RSDK-3015 - Fix streamStatus

### PR DESCRIPTION
**Pull request by cheukt**
status stream used to not work at all, reworked so it works? maybe.

pretty open to suggestions as to where I should be putting the new class, and its name, but all it is is a concrete impl of ResponseStream leveraging the existing EventDispatcher class.

tested using the below block
```
  let stream = robot.streamStatus([], new Duration().setNanos(500_000_000));
  stream.on('data', (response: VIAM.robotApi.Status[]) => {
    console.log(response);
  });
  stream.on('status', (newStatus?: { details: unknown }) => {
    console.error('error streaming robot status', newStatus);
  });
  stream.on('end', () => {
    console.error('done streaming robot status');
  });
```

### CODE REVIEW

File: src/robot/client.ts [hunk 1]
```diff
@@ -589,15 +591,29 @@ export class RobotClient implements Robot {
...
     request.setResourceNamesList(resourceNames);
     request.setEvery(duration);
-    const response = await promisify<
-      proto.StreamStatusRequest,
-      proto.StreamStatusResponse
-    >(robotService.streamStatus.bind(robotService), request);
-    return response.getStatusList();
+
+    const statusStream = robotService.streamStatus(request);
+    if (statusStream === null) {
```
Comment (maximpertsov) [inline src/robot/client.ts:604]:
```suggestion
    if (!statusStream) {
```

i'm assuming we want this to behave similar to the logic we have in `promisify` right?

**Review by maximpertsov — CHANGES_REQUESTED** [review]
This seems okay - just some minor changes.

In addition, let's codify your testing example as a unit test:

```ts
let stream = robot.streamStatus([], new Duration().setNanos(500_000_000));
  stream.on('data', (response: VIAM.robotApi.Status[]) => {
    console.log(response);
  });
  stream.on('status', (newStatus?: { details: unknown }) => {
    console.error('error streaming robot status', newStatus);
  });
  stream.on('end', () => {
    console.error('done streaming robot status');
  });
```

File: src/robot/client.ts (hunk 1 above)
Comment (cheukt) [reply src/robot/client.ts:604]: ya

**Comment by cheukt** [comment]
@maximpertsov this was tested using the sample code from app and adding the bit I wrote above

File: src/robot.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-export type { Robot } from './robot/robot';
+export type { Robot, RobotStatusStream } from './robot/robot';
```
Comment (maximpertsov) [inline src/robot.ts:1]: we should export this in `src/main.ts` as well so users have access to this type

**Pull request merged**